src/DynamoCore/Graph/Nodes/CodeBlockNode.cs

```diff
@@ -604,12 +618,14 @@ public override ProtoCore.Type GetTypeHintForOutput(int index)
 
         #region Private Methods
 
-        internal void ProcessCodeDirect()
+        internal delegate void ProcessCodeDelegate(ref string errorMessage, ref string warningMessage);
```

Comment:
should this actually use `out` to be explicit about the purpose of these parameters? Is there any time that these params are meaningful before the handler is invoked? (ie do we ever pass anything useful into the handler?)

Reply:
Yes, I think it makes sense for them to be `out` instead of `ref`. Will make the change.